src/integrationTest/java/uk/gov/hmcts/reform/cdapi/RetrieveCategoriesIntegrationTest.java

```diff
@@ -48,6 +49,18 @@ void shouldRetrieveCategoriesWithAllParamsWithStatusCode200()
         responseVerification(response);
     }
 
+    @Test
+    void shouldRetrieveCategoriesWithServiceIdWithChildNodes()
+        throws JsonProcessingException {
+        final var response = (Categories)
+            commonDataApiClient.retrieveCaseFlagsByServiceId("HearingChannel?serviceId=BBA3"
+                                                                 + "&isChildRequired=Y",
+                                                             Categories.class, path
+            );
+        assertNotNull(response);
+        assertEquals(4, response.getListOfCategory().size());
+        responseVerification(response.getListOfCategory().get(0).getChildNodes().get(0));
```

Comment:
pls add a assertion to check how many expected child nodes

Reply:
Done @rajkatla-hmcts